doc/Uplink.xml

```diff
@@ -180,15 +192,64 @@
+          Phase 2 and 3 depend on the protocol being used and are detailed below.
+          In the first phase the device acts as TLS client that connects to the cloud service.
+          Details of the TCP and TLS exchange are out of scope of this specification.
+        </para>
+        <section>
+          <title>Native h2c-reverse</title>
+          <para> <!-- HTTP/2 normal -->
+            The second phase includes the connection upgrade to HTTP/2 h2c-reverse protocol which is confirmed by the cloud service with a 101 HTTP response.
+            The third phase of the connection than fully complies to an HTTP2 connection as if it were initiated from the cloud service.
+            Please note that the figure only shows the most relevant packets.
+          </para>
+          <figure xml:id="_Ref505091745">
+            <title>Connection initiation from the device</title>
+            <mediaobject>
+              <imageobject>
+                <imagedata fileref="media/Uplink/image4.svg" contentwidth="128mm" />
+              </imageobject>
+            </mediaobject>
+          </figure>
+          <para>This protocol shall be used when the uri protocol states 'https'. A device signaling
+            support for https via its Protocol capability shall support native h2c-reverse.</para>
+        </section>
+        <section>
+          <title>WebSocket h2c-reverse</title>
+          <para>
+            <!-- WebSocket normal --> The second phase includes the connection upgrade to a
+            WebSocket with the reverse HTTP2 subprotocol which is confirmed by the cloud service
+            with a 101 HTTP response. The third phase of the connection then corresponds to an HTTP2
+            connection as if it were initiated from the cloud services, using WebSocket binary
+            frames. Please note that the figure only shows the most relevant packets. You may
+            provide an access token by adding a query parameter to the WebSocket URI as defined by
+            [RFC6750] Section 2.3. </para>
+          <figure xml:id="_Ref484594140">
+            <title>Connection initiation over websocket</title>
+            <mediaobject>
+              <imageobject>
+                <imagedata fileref="media/Uplink/image5.svg" contentwidth="128mm" />
+              </imageobject>
+            </mediaobject>
+          </figure>
+          <para>This protocol shall be used when the uri protocol states 'wss'. A device signaling
+            support for wss via its Protocol capability shall support h2c-reverse over
+            WebSocket.</para>
+        </section>
+        <section>
+          <title>HTTP/2 Frames</title>
+          <para>Once an http/2 connection has been established the communication parameters are negotiated as specified by RFC 7540 so that the uplink can be used to exchange frames between the remote client and the local service.</para>
```

Comment:
Update "an" to "a".

Reply:
According to my knowledge HTTP reads as "agettp" and hence phonetically starts with a noun ;-)

Reply:
English is full of contradictions but in answer to your question;

Reference https://owl.purdue.edu/owl/general_writing/grammar/using_articles.html
Copied directly from this article. **I am not shouting so please do not take offence.**
USING A OR AN DEPENDS ON THE SOUND THAT BEGINS THE NEXT WORD. SO...
a + singular noun beginning with a consonant: a boy; a car; a bike; a zoo; a dog
an + singular noun beginning with a vowel: an elephant; an egg; an apple; an idiot; an orphan
a + singular noun beginning with a consonant sound: a user (sounds like 'yoo-zer,' i.e. begins with a consonant 'y' sound, so 'a' is used); a university; a unicycle
an + nouns starting with silent "h": an hour
a + nouns starting with a pronounced "h": a horse
In some cases where "h" is pronounced, such as "historical," you can use an. However, a is more commonly used and preferred.

Reply:
```suggestion
          <para>Once a http/2 connection has been established the communication parameters are negotiated as specified by RFC 7540 so that the uplink can be used to exchange frames between the remote client and the local service.</para>
```